mqtt/mqtt-broker/src/state_change.rs

```diff
@@ -100,15 +103,19 @@ mod tests {
         state_change::{StateChange, STATE_CHANGE_QOS},
         subscription::{Subscription, TopicFilter},
         tests::peer_addr,
-        Auth, AuthId, BrokerConfig, ClientId, ConnReq, SessionConfig,
+        Auth, AuthId, BrokerConfig, ClientId, ClientInfo, ConnReq, SessionConfig,
     };
 
     #[test]
     fn test_subscriptions() {
         let expected_id: ClientId = "Session".into();
+        let client_info = &ClientInfo::new(
```

Comment:
IMO it's a bit confusing to create a ref. Create a value and pass it as a ref where needed.
```suggestion
        let client_info = ClientInfo::new(
```